extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -15,4 +32,36 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols() {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true, GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void registerVertxGraphqlUI(VertxGraphqlRecorder recorder,
+            BuildProducer<NativeImagePackageResourceBuildItem> nativeResourcesProducer, VertxGraphqlUiConfig config,
+            LaunchModeBuildItem launchMode, BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints,
+            BuildProducer<RouteBuildItem> routes) {
+        boolean includeVertxGraphqlUi = launchMode.getLaunchMode().isDevOrTest() || config.alwaysInclude;
+
+        if (!includeVertxGraphqlUi) {
+            return;
+        }
+
+        if ("/".equals(config.path)) {
+            throw new ConfigurationError(
+                    "quarkus.vertx-graphql-ui.path was set to \"/\", this is not allowed as it blocks the application from serving anything else.");
```

Comment:
I think I would have preferred something like:
```
quarkus.vertx-graphql.ui.path
```
with the config root being the name of the extension and `ui` being a config group. It's far more consistent with the rest of Quarkus.

Reply:
As discussed on Zulip, let’s do it :-)